sdk/core/src/main/java/org/apache/mesos/offer/constrain/AgentRule.java

```diff
@@ -34,52 +40,162 @@ public String toString() {
         return String.format("AgentRule{agentId=%s}", agentId);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        return EqualsBuilder.reflectionEquals(this, o);
```

Comment:
Are we expecting to run our JVM in that configuration? A hand-coded check risks going out of sync when new members are added.